src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAO.scala

```diff
@@ -101,7 +101,7 @@ class ElasticSearchDAO(servers:Seq[Authority], indexName: String) extends Search
   override def findDocuments(criteria: LibrarySearchParams) : LibrarySearchResponse = {
     val searchStr =
     criteria.searchTerm match {
-      case None => ESQuery(new ESMatchAll).toJson.compactPrint
+      case None | Some("") => ESQuery(new ESMatchAll).toJson.compactPrint
```

Comment:
are you concerned about whitespace-only strings, e.g. what if the user enters "   "?

You could do:
```
case None | Some(x) if x.trim == ""
```